src/test/ui/or-patterns/basic-switch.rs

```diff
@@ -0,0 +1,40 @@
+// failure-status: 101
+// rustc-env:RUST_BACKTRACE=0
+// normalize-stderr-test "note: rustc 1.* running on .*" -> "note: rustc VERSION running on TARGET"
+// normalize-stderr-test "note: compiler flags: .*" -> "note: compiler flags: FLAGS"
+#![feature(or_patterns)]
+#![feature(exclusive_range_pattern)]
+
+#[derive(Debug, PartialEq)]
+enum MatchArm {
+    Arm(usize),
+    Wild
+}
+
+#[derive(Debug)]
+enum Foo {
+    Bar(usize),
+    Baz,
+    Qux
+}
+
+#[derive(Debug)]
+enum Wrapper {
+    ContainsFoo(Foo),
+    Empty,
+}
+
+fn test_wrapper(x: Wrapper) -> MatchArm {
+    match x {
+        // most simple case
+        Wrapper::ContainsFoo(Foo::Baz | Foo::Qux) => MatchArm::Arm(0),
```

Comment:
A `Switch` on variants currently causes and ICE.

Reply:
Added basic support for matching on an `enum`. See the added test that panics and the FIXME below for cases that still fail or panic.